File: JavaScript/8js 4/take.js
/*
|--------------------------------------------------------------------------
| Enumerable Magic #25 - Take the First N Elements
|--------------------------------------------------------------------------
|
*/

function take(arr, n) {
   return arr.slice(0, n);
}


function take(arr, n) {
   return arr.reduce((acc, rec, index) => (index < n ? [...acc, rec] : acc), []);
}


function take(arr, n) {
   return arr.filter((elem, i) => i < n);
}


console.log(take([0, 1, 2, 3, 5, 8, 13], 3), [0, 1, 2]);

File: Python/8py 4/take.py
"""
|--------------------------------------------------------------------------
| Enumerable Magic #25 - Take the First N Elements
|--------------------------------------------------------------------------
|
"""

def take(arr,n):
   return arr[0:n]


take = lambda _,__: _[:__]


def take(arr,n):
   return arr[:n]


print(take([0, 1, 2, 3, 5, 8, 13], 3), [0, 1, 2])